crates/circuit/src/dag_circuit.rs

```diff
@@ -2416,78 +2417,90 @@ def _format(operand):
-        //     return node
-        //
-        // new_node = copy.copy(node)
-        // new_node.op = op
-        // self._multi_graph[node._node_id] = new_node
-        // if op.name != node.op.name:
-        //     self._increment_op(op)
-        //     self._decrement_op(node.op)
-        // return new_node
-        todo!()
+    ) -> PyResult<Py<PyAny>> {
+        let py = op.py();
+        let new_op = convert_py_to_operation_type(py, op.clone().unbind())?;
+        // if either node is a control-flow operation, propagate_condition is ignored
+        let new_condition = match propagate_condition
+            && !new_op.operation.control_flow()
+            && !node.instruction.operation.control_flow()
+        {
+            true => {
+                // if new_op has a condition, the condition can't be propagated from the old node
+                if new_op.condition.is_some() {
+                    return Err(DAGCircuitError::new_err(
+                        "Cannot propagate a condition to an operation that already has one.",
+                    ));
+                } else {
+                    match &node.instruction.extra_attrs {
+                        Some(extra_attrs) => &extra_attrs.condition,
+                        None => &None,
+                    }
+                }
+            }
+            // if propagate_condition is false or ignored, the new_op condition (if any) will be used
+            false => &new_op.condition,
+        };
+        let old_id = node.as_ref().node.unwrap();
+        match self.dag.clone().node_weight(old_id) {
+            Some(NodeType::Operation(old_packed)) => {
+                let new_weight = NodeType::Operation(PackedInstruction::new(
+                    new_op.operation,
+                    old_packed.qubits_id,
+                    old_packed.clbits_id,
+                    new_op.params,
+                    new_op.label,
+                    new_op.duration,
+                    new_op.unit,
+                    new_condition.clone(),
+                    #[cfg(feature = "cache_pygates")]
+                    Some(op.unbind()),
+                ));
+                // using self.dag.contract_nodes to replace a single node. Is there a better method
+                // to do it?
+                let new_node = self
+                    .dag
+                    .contract_nodes(Some(old_id).into_iter(), new_weight.clone(), false)
+                    .unwrap();
+                let new_name = match &new_weight {
+                    NodeType::Operation(new_packed) => new_packed.op.name(),
+                    _ => unreachable!(),
+                };
+                self.decrement_op(old_packed.op.name().to_string());
+                self.increment_op(new_name.to_string());
+
+                if !inplace {
+                    Ok(self.get_node(py, new_node)?)
+                } else {
+                    let new_op = convert_py_to_operation_type(py, op.clone().unbind())?;
```

Comment:
It seems to be not possible to clone `new_op`, and it is being moved into `new_weight`, so I am re-generating it here for lack of a better solution.